### PR TITLE
Create SentinelSearch.php

### PR DESCRIPTION
**Pull request by Abhishek-Pashte**
Added **Sentinel Search algorithm** in Searches algorithms

### CODE REVIEW

**Review by darwinz — CHANGES_REQUESTED** [review]
Thank you for contributing.  Please add this algorithm to the [Directory](https://github.com/TheAlgorithms/PHP/blob/master/DIRECTORY.md) and add unit tests.  See [CONTRIBUTING](https://github.com/TheAlgorithms/PHP/blob/master/CONTRIBUTING.md) if you have any questions.

**Comment by Abhishek-Pashte** [comment]
Required changes has been made

**Comment by Abhishek-Pashte** [comment]
Required changes has been made. Pls review those changes

**Comment by darwinz** [comment]
Thanks for making those changes.  Please fix the linting errors that came up in the code-style workflow checks.

**Comment by darwinz** [comment]
@Abhishek-Pashte , looks like we still have some linting errors

File: Searches/SentinelSearch.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    Output : -
+    Returns index of element if found, else -1
+*/
+function SentinelSearch($list, $target)
+{
+    //Length of array
+    $len = sizeof($list);
+
+    //Store last element of array
+    $lastElement = $list[$len-1];
```
Comment (darwinz) [inline Searches/SentinelSearch.php:17]:
```suggestion
    $lastElement = $list[$len - 1];
```

File: Searches/SentinelSearch.php [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    //Put target at the last position of array known as 'Sentinel'
+    if ($lastElement == $target)
```
Comment (darwinz) [inline Searches/SentinelSearch.php:20]:
```suggestion
    if ($lastElement == $target) 
```

File: Searches/SentinelSearch.php [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    {
+        return ($len - 1);
+    }
```
Comment (darwinz) [inline Searches/SentinelSearch.php:23]:
```suggestion
    }
```

File: Searches/SentinelSearch.php [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+    //Store last element of array
+    $lastElement = $list[$len-1];
+
+    //Put target at the last position of array known as 'Sentinel'
+    if ($lastElement == $target)
+    {
+        return ($len - 1);
+    } 
+
```
Comment (darwinz) [inline Searches/SentinelSearch.php:24]:
```suggestion

```

File: Searches/SentinelSearch.php [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    $list[$len - 1] = $target;
+
+    //Initialize variable to traverse through array
+    $i=0;
+
+    //Traverse through array to search target
+    while ($list[$i] != $target)
+    {
+        $i++;
+    }
```
Comment (darwinz) [inline Searches/SentinelSearch.php:35]:
```suggestion
    $i = 0;

    //Traverse through array to search target
    while ($list[$i] != $target) 
    {
        $i++;
    }
```

File: Searches/SentinelSearch.php [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    $list[$len - 1] = $lastElement;
+
+    //If i in less than length, It means element is present in array
+    if ($i < ($len - 1))
+        return $i;
+    // Else element is not present return -1
+    else 
+        return -1;
+}
+?>
```
Comment (darwinz) [inline Searches/SentinelSearch.php:46]:
```suggestion
    if ($i < ($len - 1)) {
        return $i;
    }
    // Else element is not present return -1
    else {
        return -1;
    }
}
```

File: Searches/SentinelSearch.php [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+    //If i in less than length, It means element is present in array
+    if ($i < ($len - 1)) {
+        return $i;
+    }
+    // Else element is not present return -1
+    else {
+        return -1;
+    }
+}
```
Comment (darwinz) [inline Searches/SentinelSearch.php:46]:
```suggestion
{
    //Length of array
    $len = sizeof($list);

    //Store last element of array
    $lastElement = $list[$len - 1];

    //Put target at the last position of array known as 'Sentinel'
    if ($lastElement == $target) {
        return ($len - 1);
    }
    //Put target at last index of array
    $list[$len - 1] = $target;

    //Initialize variable to traverse through array
    $i = 0;

    //Traverse through array to search target
    while ($list[$i] != $target) {
        $i++;
    }
    //Put last element at it's position
    $list[$len - 1] = $lastElement;

    //If i in less than length, It means element is present in array
    if ($i < ($len - 1)) {
        return $i;
    } else {
        return -1;
    }
}
```

**Review by darwinz — APPROVED** [review]
Thanks!

**Pull request merged**